File: backend/src/main/java/com/kishan/backend/auth/service/AuthService.java
package com.kishan.backend.auth.service;

import com.kishan.backend.auth.dto.LoginRequest;
import com.kishan.backend.auth.dto.LoginResponse;
import com.kishan.backend.auth.dto.RegisterRequest;
import com.kishan.backend.auth.dto.RegisterResponse;

/**
 * Service interface defining business logic operations related to user authentication and management.
 */
public interface AuthService {
    
    /**
     * Registers a new user in the system.
     * Checks for email duplicates, hashes the password, and persists the user record.
     *
     * @param request the registration details
     * @return the details of the successfully registered user
     * @throws com.kishan.backend.auth.exception.EmailAlreadyExistsException if email is already taken
     */
    RegisterResponse register(RegisterRequest request);

    /**
     * Authenticates a user with email and password and returns a login response.
     *
     * @param request the login request details
     * @return the login response containing the token and user details
     */
    LoginResponse login(LoginRequest request);
}

File: frontend/src/services/authService.js
import apiClient from '../api/axios';

const AUTH_ENDPOINTS = {
  REGISTER: '/auth/register',
  LOGIN: '/auth/login',
};

export const registerUser = async ({ email, password, name }) => {
  const response = await apiClient.post(AUTH_ENDPOINTS.REGISTER, {
    email,
    password,
    name,
  });
  return response.data;
};

export const loginUser = async ({ email, password }) => {
  const response = await apiClient.post(AUTH_ENDPOINTS.LOGIN, {
    email,
    password,
  });
  return response.data;
};
